gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/AbstractChannelizer.java

```diff
@@ -200,6 +211,35 @@ private Authenticator createAuthenticator(final Settings.AuthenticationSettings
         }
     }
 
+    private Authorizer createAuthorizer(final Settings.AuthorizationSettings config) {
+        final String authorizerClass = config.authorizer;
+        if (null == authorizerClass) {
+            return null;
+        }
+        try {
+            final Class<?> clazz = Class.forName(authorizerClass);
+            final Authorizer authorizer = (Authorizer) clazz.newInstance();
+            authorizer.configure(config.config);
+            return authorizer;
+        } catch (Exception ex) {
+            logger.warn(ex.getMessage());
+            throw new IllegalStateException(String.format("Could not create/configure Authorizer %s", authorizer), ex);
+        }
+    }
+
+    private AuthenticatorAuthorizer createAuthenticatorAuthorizer(final Settings settings) {
+        try {
+            final Class<?> clazz = Class.forName(settings.authentication.authenticator);
+            final AuthenticatorAuthorizer authenticatorAuthorizer = (AuthenticatorAuthorizer) clazz.newInstance();
+            authenticatorAuthorizer.setup(settings.authentication.config);
+            authenticatorAuthorizer.configure(settings.authorization.config);
```

Comment:
`setup()` for authentication and `configure()` for authorization - could we simply unify these and go with `setup()` for both? As long as the two methods have the same name/signature then it should look like one method from an inheritance perspective, no?

Reply:
This refers to my earlier question (that you possibly overlooked) whether we should have this AuthenticatorAuthorizer interface anyway. Only reason for its existence is for the Authorizer to have easy access to any user details not present in the AuthenticatedUser (QwentB's request). The user or provide could realize this in an alternative way by building or wrapping an Authenticator implementation into a singleton.
Agreed, that the setup/configure thing is too ugly too keep, but is also ugly to not give the AuthentionSetting and AuthorizationSetting their own config map. Not relevant if we do away with the AuthenticatorAuthorizer.

Reply:
I'm +0 with the `AuthenticatorAuthorizer` - whatever you decide on that is fine by me. I'm more concerned in just naming those methods the same thing.  I probably shouldn't have written that the methods should have the same signature I guess...just the same name. I guess we would go with `setup()` which is somewhat at odds with all our other things which use `configure()` but meh...it's fine